docs/src/pages/ComponentArchitecture.mdx

```diff
@@ -0,0 +1,97 @@
+For variants that are expected in the overall component library, state hooks should be reused and a new component simply
+created with a different use-case.
+
+It is important to have consistent accessibility, and passing in a new accessibility model might create major
+challenges to keep our components accessible, and also predictable. If a particular component needs to have multiple
+accessibility models (keyboarding, attributes), it should make that decision based on a single prop with a fixed set
+of implementation options (i.e. a `accessibilityMode` prop that accepts a fixed set of string values as input).
+
+# Themed components
+
+The control library should ship a set of unopinionated components (i.e. base components) but also components which are
+pre-scaffolded with Fluent design opinions (themed components). Fluent will be opinionated but customizable.
+
+#### Themed components are the glue that ties together context, theming, and the base component. There are several potential implementations (`createComponent`, other HoCs), but serve as the higher level abstraction that brings only the necessary pieces to a consumer's application bundle.
+
+Themed components should pass a set of classes to base components. In practice, this will mean pulling
+together information about theming and styling from context, and ensuring the correct CSS is rendered on
+the page.
+
+#### The themed layer is the preferred layer to add hard graph edges. Each themed control should be intuitive to use for a developer, and should come with "batteries included".
+
+Each themed component that relies on sub-components should specify which components to render through
+the slots prop.
+
+The slots prop must be accessible in several means to ensure that a "recursive component" can be accurately described.
+
+#### We want to offer a delightful first-run experience, and bundle size will not be negatively impacted by including the default theme.
+
+Each component should render a known-good theme out of the box when used without a provider.
+
+#### As a component library cannot anticipate all needs to applications' design systems, allowances should be made to extend the design system while maintaining rich levels of type safety.
+
+The HoC used to glue base components to the rest of the system should also provide a facility
+to define a variant. A variant is defined as an additional prop on the created component which has
+the ability to adjust the look and feel of the component.
+
+#### Themed components pass through unhandled but filtered props, which should be rendered on the `root` of a base component
+
+Each themed control should accept any number of props that will be rendered by the base control's root slot.
+The base control should leverage a platform library's function to pass all relevant native props to appropriate
+controls whenever available.
```

Comment:
This point `#### Themed components pass ...` and explainer is great.  Let's keep but strip the details about theme component etc.  Keep it to how we handle known/safe props.